app/src/main/java/com/aaronjwood/portauthority/async/ScanHostsAsyncTask.java

```diff
@@ -40,17 +40,18 @@ public ScanHostsAsyncTask(MainAsyncResponse delegate) {
     protected Void doInBackground(String... params) {
         String ip = params[0];
         String parts[] = ip.split("\\.");
+        int hosts = Integer.decode(params[1]);
 
         ExecutorService executor = Executors.newCachedThreadPool();
 
         int SCAN_THREADS = 8;
-        int chunk = (int) Math.ceil((double) 255 / SCAN_THREADS);
+        int chunk = (int) Math.ceil((double) hosts / SCAN_THREADS);
```

Comment:
I think what I'll do here after this is merged in is scale up the number of threads to some sort of maximum based on the number of hosts to scan. If someone has 1000's of hosts to scan it'll take way longer than it should :)

I can open a separate issue for this if you want.